repentogon/resources/scripts/main_ex.lua

```diff
@@ -1407,6 +1412,109 @@ function _RunPostPickupSelection(callbackID, param, pickup, variant, subType, ..
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `1` (number expected, got "..type1..")")
+				elseif math.type(ret[1]) ~= "integer" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `1` (number has no integer representation)")
+				elseif type2 ~= "Vector" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `2` (Vector expected, got "..type2..")")
+				elseif type3 ~= "boolean" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `3` (boolean expected, got "..type3..")")
+				else
+					recentRet = ret
+					duration = ret[1]
+					extraParam1 = ret[2]
+					extraParam2 = ret[3]
+				end
+			elseif status == StatusEffect.SLOWING then
+				local type1 = type(ret[1])
+				if type1 == "userdata" then type1 = GetMetatableType(ret[1]) end
+				local type2 = type(ret[2])
+				if type2 == "userdata" then type2 = GetMetatableType(ret[2]) end
+				local type3 = type(ret[3])
+				if type3 == "userdata" then type3 = GetMetatableType(ret[3]) end
+
+				if type1 ~= "number" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `1` (number expected, got "..type1..")")
+				elseif math.type(ret[1]) ~= "integer" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `1` (number has no integer representation)")
+				elseif type2 ~= "number" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `2` (number expected, got "..type2..")")
+				elseif type3 ~= "Color" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `3` (Color expected, got "..type3..")")
+				else
+					recentRet = ret
+					duration = ret[1]
+					extraParam1 = ret[2]
+					extraParam2 = ret[3]
+				end
+			elseif status == StatusEffect.BURN or status == StatusEffect.POISON then
+				local type1 = type(ret[1])
+				if type1 == "userdata" then type1 = GetMetatableType(ret[1]) end
+				local type2 = type(ret[2])
+				if type2 == "userdata" then type2 = GetMetatableType(ret[2]) end
+
+				if type1 ~= "number" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `1` (number expected, got "..type1..")")
+				elseif math.type(ret[1]) ~= "integer" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `1` (number has no integer representation)")
+				elseif type2 ~= "number" then
+					logError(callbackID, callback.Mod.Name, "bad return type for table value with key `2` (number expected, got "..type2..")")
+				else
+					recentRet = ret
+					duration = ret[1]
+					extraParam1 = ret[2]
+				end
+			else
+				logError(callbackID, callback.Mod.Name, "bad return type (table not expected for status)")
+			end
+		elseif type(ret) == "number" then
+			recentRet = ret
+		end
+	end
+
+	return recentRet
```

Comment:
You might need to account for the situation where one callback returns a table, then a later callback just returns an integer. You still need to maintain the extra params from the table return.

You can either update the table instead of replacing it if someone returns a number after a table was returned, or you could choose to just always return a table like `return {duration, extraParam1, extraParam2}` so long as nobody returns false.

Similarly make sure that someone returning true doesn't overwrite the table stored as recentRet